### PR TITLE
js: Refactor initializing a bundle

### PR DESCRIPTION
**Pull request by mstoykov**
DRYing the initialization of a bundle and adding a test for previously not tested archiving of an archive

### CODE REVIEW

**Comment by mstoykov** [comment]
This is an extract from a much larger js refactor.

It just happens to be mostly self contained and a lot less ... controversial then the remaining changes so I decided to pull it out.

The test is because I *did* break this originally, but currently can't remember how or what I did wrong. But still a good idea to have a small test for this.

File: js/bundle.go [hunk 1]
```diff
@@ -63,6 +63,13 @@ func (bi *BundleInstance) getExported(name string) goja.Value {
 // NewBundle creates a new bundle from a source file and a filesystem.
 func NewBundle(
 	piState *lib.TestPreInitState, src *loader.SourceData, filesystems map[string]afero.Fs,
+) (*Bundle, error) {
+	return newBundle(piState, src, filesystems, lib.Options{}, true)
+}
+
+func newBundle(
+	piState *lib.TestPreInitState, src *loader.SourceData, filesystems map[string]afero.Fs,
+	options lib.Options, updateOptions bool, // TODO: try to figure out a way to not need both
```
Comment (codebien) [inline js/bundle.go:72]: Out of curiosity, does the larger refactor help here?

**Review by codebien — APPROVED** [review]
LGTM

File: js/bundle.go (hunk 1 above)
Comment (mstoykov) [reply js/bundle.go:72]: Not really, but IMO this is on the "small problems" list of this codebase :sweat_smile:

Comment (mstoykov) [reply js/bundle.go:72]: Arguably I can drop the boolean if I extract 

```golang
    err = bundle.getExports(piState.Logger, rt, updateOptions)
	if err != nil {
		return nil, err
	}
```

in each caller - but I am less happy with this idea then with providing the options and a boolean. And unfortunately we *need* the options for other reasons so not providing them is not possible :( .

File: js/runner_test.go [hunk 2]
```diff
@@ -2745,3 +2745,35 @@ exports.default = () => {
 		})
 	}
 }
+
+func TestArchivingAnArchiveWorks(t *testing.T) {
```
Comment (imiric) [inline js/runner_test.go:2749]: This test passes on `master` as well.

Doesn't it need to assert that the options _weren't_ updated after the second `NewFromArchive()` call?

**Review by imiric — COMMENTED** [review]
LGTM, just a test comment.

But I would like to see the reason explaining _why_ this is done; both in the commit message and the PR description. GH will populate the PR description with the commit message if there's only one commit, so you only need to type it once.

If this is avoiding a second archive from updating options, then it sounds more like a fix than a refactor. But where is this an issue? All of this should be explained. :)

File: js/runner_test.go (hunk 2 above)
Comment (mstoykov) [reply js/runner_test.go:2749]: It is passing on master as it isn't fixing anything it is just a test that I broke on the original writing of this.  If this commit was fixing somethign it wouldn't be a refactor ;) 

And the point was that I broke the whole *making* of the archive from the archive. Which is also what the names suggests

**Comment by mstoykov** [comment]
> why this is done 

it is done as the logic for how the bundle was initilizied was basically repeated in two places and it is now in a single one. I don't think the commit message can be any more consice without going over the 60 character limit and I really don't think that for a refactor of this kind I need to write a small novel on why we are doing it.

File: js/runner_test.go (hunk 2 above)
Comment (imiric) [reply js/runner_test.go:2749]: OK, makes sense.

**Review by imiric — APPROVED** [review]
> it is done as the logic for how the bundle was initilizied was basically repeated in two places and it is now in a single one.

I actually didn't notice that, so a more detailed commit message would've helped.

> I don't think the commit message can be any more consice without going over the 60 character limit and I really don't think that for a refactor of this kind I need to write a small novel on why we are doing it.

It's not about writing a small novel. Just mentioning that we're DRYing things and adding a test for a previously broken feature would answer the "why". All commit messages should answer that; a "refactor" doesn't say anything.

**Pull request merged**